.github/workflows/site.yaml

```diff
@@ -1,13 +1,48 @@
 name: Site
+
 on:
-  - pull_request
+  push:
+    tags:
+      - "v*.*.*"
+  pull_request:
+
+env:
+  UV_SYSTEM_PYTHON: true
+
+permissions:
+  id-token: write
+  pages: write
+
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
-env:
-  UV_SYSTEM_PYTHON: true
+
 jobs:
+  site:
```

Comment:
FYI: 

Some of the R packages publish their latest `main` docs at "/dev/" paths, like https://rstudio.github.io/rsconnect/dev/, while publishing their released docs at https://rstudio.github.io/rsconnect/.

They don't generally have per-PR previews.